Cargo.toml

```diff
@@ -53,13 +53,24 @@ version = "1.0"
 default-features = false
 features = ["std"]
 
+[dependencies.quickcheck]
+optional = true
+version = "0.8"
+default-features = false
+
+[dependencies.quickcheck_macros]
+optional = true
+version = "0.8"
+default-features = false
+
 [dev-dependencies.serde_test]
 version = "1.0"
 
 [features]
 default = ["std"]
 i128 = ["num-integer/i128", "num-traits/i128"]
 std = ["num-integer/std", "num-traits/std"]
+quick_check = ["quickcheck","quickcheck_macros"]
```

Comment:
While the attribute looks nice, I'm not keen on having this affect the public features. Let's just stick to the `quickcheck!` macro.

Reply:
Yeah, I suspected you might not like this. Is there any better way to group optional dependencies? I was looking for something that would make it so you could select either `quickcheck` and `quickcheck_macros` or nothing, but found no way to accomplish that.

What do you mean by affect public features? If we don't re-export `quickcheck_macros`, it doesn't affect public features, right?

Reply:
By public features, I mean everything that would be valid for someone to add in their:

```toml
[dependencies.num-bigint]
version = "0.2"
features = [...]
```

All optional dependencies become named features, in addition to the explicit crate features, so here we would have all three choices available to the user: `quickcheck`, `quickcheck_macros`, and `quick_check`.

The most convenient solution for us would be if `quickcheck` had a feature to reexport the macros directly, just like `serde` does with its `derive` feature.

Reply:
Another downside is that we only need the macros for testing, but it will get forced in the build for all of our `quickcheck`-enabled users too. This is unfortunate because `syn` is pretty slow to build.

Reply:
Maybe just forget about `quick_check`? Users can enable `quickcheck` alone to get the implementations. Then we can enable `quickcheck_macros` when we run the tests, and gate on that `cfg` in the test file, but as a crate feature that will be useless for users.
(Enable this if you want to make your build slower! :wink: )